lib/internal/errors.js

```diff
@@ -86,6 +87,51 @@ function makeNodeError(Base) {
   };
 }
 
+function makeNodeError2(Base, key) {
+  return class NodeError extends Base {
+    constructor(...args) {
+      super(message(key, args));
+    }
+
+    get name() {
+      return `${super.name} [${key}]`;
+    }
```

Comment:
In this case we might also want to create a `static get name()` so that the class itself can be disambiguated from other classes of this type?

Reply:
Or even better, `NodeError.name = ...`.

Reply:
I thought about this a few times now and I decided not to do that out of the following reasons:

1) We only use the class internal and do not expose it directly.
2) The main class has sub classes as properties and those should make it easy to identify the correct one.
3) If we add the name, the original `NodeError` name will not be there and I think it should be.